tests/general/signals.c

```diff
@@ -0,0 +1,88 @@
+#include <signal.h>
+#include <string.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+
+// Check for C-standard macros.
+#ifndef SIGABRT
+#error SIGABRT not defined
+#endif
+#ifndef SIGFPE
+#error SIGFPE not defined
+#endif
+#ifndef SIGILL
+#error SIGILL not defined
+#endif
+#ifndef SIGINT
+#error SIGINT not defined
+#endif
+#ifndef SIGSEGV
+#error SIGSEGV not defined
+#endif
+#ifndef SIGTERM
+#error SIGTERM not defined
+#endif
```

Comment:
Are these really necessary?    Seems like just using them should be fine.

Reply:
I've now extended the test to actually use all these.